designer/app.py

```diff
@@ -760,12 +761,8 @@ def _perform_new(self, *args):
 
         self.close_popup()
 
-        new_proj_dir = os.path.join(get_config_dir(),
-                                    constants.NEW_PROJECT_DIR_NAME)
-        if os.path.exists(new_proj_dir):
-            shutil.rmtree(new_proj_dir)
-
-        os.mkdir(new_proj_dir)
+        new_proj_dir = tempfile.mkdtemp(prefix='designer_',
+                                        dir=get_config_dir())
```

Comment:
I've been working on a option 1 (#294) because I have noticed my testing projects have still not been removed in "C:\Users\USERNAME\AppData\Local\Temp" (after more than 16 days).

However I am not sure on this solution now, as @manthansharma states it gives the user a chance to save data. 

Also there is no guarantee the folders will be removed: `The user of mkdtemp() is responsible for deleting the temporary directory and its contents when done with it.` (from: [https://docs.python.org/3/library/tempfile.html](https://docs.python.org/3/library/tempfile.html)